generator/src/lib.rs

```diff
@@ -39,19 +39,19 @@ pub fn derive_parser(input: TokenStream, include_grammar: bool) -> TokenStream {
     let (name, generics, content) = parse_derive(ast);
 
     let (data, path) = match content {
-        GrammarSource::File(ref path) => {
+        GrammarSource::File(path) => {
             let root = env::var("CARGO_MANIFEST_DIR").unwrap_or_else(|_| ".".into());
-            let path = Path::new(&root).join("src/").join(&path);
-            let file_name = match path.file_name() {
+            let full_path = Path::new(&root).join("src/").join(&path);
+            let file_name = match full_path.file_name() {
```

Comment:
FYI @tmandry it looks like this has broken the recent Pest 2.2.0 release (#672). While `include_str!` takes a path relative to a source file - which could live in a module like `$CARGO_MANIFEST_DIR/src/parser/<.pest file>` that exact same filename is appended here directly to `src/`, missing the `parser/` subdirectory.

Reply:
It doesn't look like it is very easy to get to the filepath of the source file that invoked a derive/proc-macro: https://users.rust-lang.org/t/proc-macro-how-to-identify-file-being-compiled/22617 / https://github.com/rust-lang/rust/issues/54725.